File: jscomp/test/string_runtime_test.js
'use strict';

var Mt = require("./mt.js");
var List = require("../../lib/js/list.js");
var Bytes = require("../../lib/js/bytes.js");
var Test_char = require("./test_char.js");
var Caml_bytes = require("../../lib/js/caml_bytes.js");

var suites_0 = [
  "caml_is_printable",
  (function (param) {
      return {
              TAG: /* Eq */0,
              _0: Test_char.caml_is_printable(/* 'a' */97),
              _1: true
            };
    })
];

var suites_1 = {
  hd: [
    "caml_string_of_bytes",
    (function (param) {
        var match = List.split(List.map((function (x) {
                    var b = Caml_bytes.caml_create_bytes(x);
                    Bytes.fill(b, 0, x, /* 'c' */99);
                    return [
                            Bytes.to_string(b),
                            Caml_bytes.bytes_to_string(Bytes.init(x, (function (param) {
                                        return /* 'c' */99;
                                      })))
                          ];
                  }), {
                  hd: 1000,
                  tl: {
                    hd: 1024,
                    tl: {
                      hd: 1025,
                      tl: {
                        hd: 4095,
                        tl: {
                          hd: 4096,
                          tl: {
                            hd: 5000,
                            tl: {
                              hd: 10000,
                              tl: /* [] */0
                            }
                          }
                        }
                      }
                    }
                  }
                }));
        return {
                TAG: /* Eq */0,
                _0: match[0],
                _1: match[1]
              };
      })
  ],
  tl: /* [] */0
};

var suites = {
  hd: suites_0,
  tl: suites_1
};

Mt.from_pair_suites("String_runtime_test", suites);

exports.suites = suites;
/*  Not a pure module */

File: jscomp/test/string_runtime_test.ml
module S = Js__Caml_string
module B = Js__Caml_bytes
(** TODO: performance could be improved, however,
    this function is not in critical Path
 *)
let suites = Mt.[
  (* "string_of_char_array", (fun _ ->
    Eq(S.caml_string_of_char_array [|'a';'b';'c'|], "abc")
                          ); *)
  "caml_is_printable", (fun _ ->
    Eq(Test_char.caml_is_printable 'a', true)
                       );
  "caml_string_of_bytes", (fun  _ ->

    let f len =
      let b = Bytes.create len in
      Bytes.fill b 0 len 'c';
      (Bytes.to_string b,  (String.init len (fun _ -> 'c'))) in
    let (a,b) =
      List.split @@ List.map (fun x -> f x ) [1000;1024;1025;4095;4096;5000;10000] in
    Eq(a,b)
                          )

]

;;Mt.from_pair_suites __MODULE__ suites
